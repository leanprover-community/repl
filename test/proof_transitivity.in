{"cmd": "example (x y z : Nat) (h1 : x = y) (h2 : y = z) : x = z := by sorry", "sorries": "individual"}

{"tactic": "exact Eq.trans h1 h2", "proofState": 0}

{"cmd": "example (f : Nat â†’ Nat) (n : Nat) (h : n = 3) : f n = f 3 := by sorry", "sorries": "individual"}

{"tactic": "exact congrArg f h", "proofState": 2}

