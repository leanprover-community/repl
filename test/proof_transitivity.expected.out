{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 62},
   "parentDecl": "_example",
   "goals": ["x y z : Nat\nh1 : x = y\nh2 : y = z\n⊢ x = z"],
   "endPos": {"line": 1, "column": 67}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Completed", "proofState": 1, "goals": []}

{"sorries":
 [{"proofState": 2,
   "pos": {"line": 1, "column": 64},
   "parentDecl": "_example",
   "goals": ["f : Nat → Nat\nn : Nat\nh : n = 3\n⊢ f n = f 3"],
   "endPos": {"line": 1, "column": 69}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"proofStatus": "Completed", "proofState": 3, "goals": []}

